In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/youtube-new/GBvideos.csv
/kaggle/input/youtube-new/MXvideos.csv
/kaggle/input/youtube-new/KRvideos.csv
/kaggle/input/youtube-new/MX_category_id.json
/kaggle/input/youtube-new/GB_category_id.json
/kaggle/input/youtube-new/US_category_id.json
/kaggle/input/youtube-new/IN_category_id.json
/kaggle/input/youtube-new/DEvideos.csv
/kaggle/input/youtube-new/KR_category_id.json
/kaggle/input/youtube-new/RU_category_id.json
/kaggle/input/youtube-new/FRvideos.csv
/kaggle/input/youtube-new/USvideos.csv
/kaggle/input/youtube-new/INvideos.csv
/kaggle/input/youtube-new/RUvideos.csv
/kaggle/input/youtube-new/CA_category_id.json
/kaggle/input/youtube-new/DE_category_id.json
/kaggle/input/youtube-new/JP_category_id.json
/kaggle/input/youtube-new/JPvideos.csv
/kaggle/input/youtube-new/FR_category_id.json
/kaggle/input/youtube-new/CAvideos.csv


### canadian dataset

In [2]:
df_ca = pd.read_csv("../input/youtube-new/CAvideos.csv")
df_ca.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40881 entries, 0 to 40880
Data columns (total 16 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   video_id                40881 non-null  object
 1   trending_date           40881 non-null  object
 2   title                   40881 non-null  object
 3   channel_title           40881 non-null  object
 4   category_id             40881 non-null  int64 
 5   publish_time            40881 non-null  object
 6   tags                    40881 non-null  object
 7   views                   40881 non-null  int64 
 8   likes                   40881 non-null  int64 
 9   dislikes                40881 non-null  int64 
 10  comment_count           40881 non-null  int64 
 11  thumbnail_link          40881 non-null  object
 12  comments_disabled       40881 non-null  bool  
 13  ratings_disabled        40881 non-null  bool  
 14  video_error_or_removed  40881 non-null  bool  
 15  de

from the general overview, a few things that stand out are:
* not all channels have a description
* trending date and publish time are listed as an object, they could be candidates for conversion to datetime format

In [3]:
df_ca.describe()

category_id         views         likes      dislikes  comment_count
count  40881.000000  4.088100e+04  4.088100e+04  4.088100e+04   4.088100e+04
mean      20.795553  1.147036e+06  3.958269e+04  2.009195e+03   5.042975e+03
std        6.775054  3.390913e+06  1.326895e+05  1.900837e+04   2.157902e+04
min        1.000000  7.330000e+02  0.000000e+00  0.000000e+00   0.000000e+00
25%       20.000000  1.439020e+05  2.191000e+03  9.900000e+01   4.170000e+02
50%       24.000000  3.712040e+05  8.780000e+03  3.030000e+02   1.301000e+03
75%       24.000000  9.633020e+05  2.871700e+04  9.500000e+02   3.713000e+03
max       43.000000  1.378431e+08  5.053338e+06  1.602383e+06   1.114800e+06

In [4]:
df_ca.head()

video_id trending_date  \
0  n1WpP7iowLc      17.14.11   
1  0dBIkQ4Mz1M      17.14.11   
2  5qpjK5DgCt4      17.14.11   
3  d380meD0W0M      17.14.11   
4  2Vv-BfVoq4g      17.14.11   

                                               title channel_title  \
0         Eminem - Walk On Water (Audio) ft. Beyoncé    EminemVEVO   
1                      PLUSH - Bad Unboxing Fan Mail     iDubbbzTV   
2  Racist Superman | Rudy Mancuso, King Bach & Le...  Rudy Mancuso   
3                           I Dare You: GOING BALD!?      nigahiga   
4        Ed Sheeran - Perfect (Official Music Video)    Ed Sheeran   

   category_id              publish_time  \
0           10  2017-11-10T17:00:03.000Z   
1           23  2017-11-13T17:00:00.000Z   
2           23  2017-11-12T19:05:24.000Z   
3           24  2017-11-12T18:01:41.000Z   
4           10  2017-11-09T11:04:14.000Z   

                                                tags     views    likes  \
0  Eminem|"Walk"|"On"|"Water"|"Aftermath/Shady/In...  17158579   787425   
1  plush|"bad unboxing"|"unboxing"|"fan mail"|"id...   1014651   127794   
2  racist superman|"rudy"|"mancuso"|"king"|"bach"...   3191434   146035   
3  ryan|"higa"|"higatv"|"nigahiga"|"i dare you"|"...   2095828   132239   
4  edsheeran|"ed sheeran"|"acoustic"|"live"|"cove...  33523622  1634130   

   dislikes  comment_count                                  thumbnail_link  \
0     43420         125882  https://i.ytimg.com/vi/n1WpP7iowLc/default.jpg   
1      1688          13030  https://i.ytimg.com/vi/0dBIkQ4Mz1M/default.jpg   
2      5339           8181  https://i.ytimg.com/vi/5qpjK5DgCt4/default.jpg   
3      1989          17518  https://i.ytimg.com/vi/d380meD0W0M/default.jpg   
4     21082          85067  https://i.ytimg.com/vi/2Vv-BfVoq4g/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   
3              False             False                   False   
4              False             False                   False   

                                         description  
0  Eminem's new track Walk on Water ft. Beyoncé i...  
1  STill got a lot of packages. Probably will las...  
2  WATCH MY PREVIOUS VIDEO ▶ \n\nSUBSCRIBE ► http...  
3  I know it's been a while since we did this sho...  
4  🎧: https://ad.gt/yt-perfect\n💰: https://atlant...

In [5]:
df_ca.nunique()

video_id                  24427
trending_date               205
title                     24573
channel_title              5076
category_id                  17
publish_time              23613
tags                      20157
views                     40170
likes                     24676
dislikes                   6241
comment_count             11172
thumbnail_link            24422
comments_disabled             2
ratings_disabled              2
video_error_or_removed        2
description               22345
dtype: int64

* there are 205 unique dates
* the number of unique values in the category section is 17. This is a candidate for conversion to a categorical variable.
* comments_disabled, ratings_disabled and video_error_or_removed are also candidates for conversion to categorical variables

In [6]:
df_ca.duplicated().sum()

0

### Great Britain

In [7]:
df_gb = pd.read_csv("../input/youtube-new/GBvideos.csv")
df_gb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 38916 entries, 0 to 38915
Data columns (total 16 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   video_id                38916 non-null  object
 1   trending_date           38916 non-null  object
 2   title                   38916 non-null  object
 3   channel_title           38916 non-null  object
 4   category_id             38916 non-null  int64 
 5   publish_time            38916 non-null  object
 6   tags                    38916 non-null  object
 7   views                   38916 non-null  int64 
 8   likes                   38916 non-null  int64 
 9   dislikes                38916 non-null  int64 
 10  comment_count           38916 non-null  int64 
 11  thumbnail_link          38916 non-null  object
 12  comments_disabled       38916 non-null  bool  
 13  ratings_disabled        38916 non-null  bool  
 14  video_error_or_removed  38916 non-null  bool  
 15  de

In [8]:
df_ca.columns == df_gb.columns

array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True])

here I verify that the columns in great britain are the same as the columns in canada

In [9]:
df_gb.describe()

category_id         views         likes      dislikes  comment_count
count  38916.000000  3.891600e+04  3.891600e+04  3.891600e+04   3.891600e+04
mean      16.827937  5.911944e+06  1.345196e+05  7.612560e+03   1.308835e+04
std        7.752728  1.900121e+07  3.499893e+05  5.095683e+04   5.066740e+04
min        1.000000  8.510000e+02  0.000000e+00  0.000000e+00   0.000000e+00
25%       10.000000  2.515272e+05  5.897000e+03  2.000000e+02   6.790000e+02
50%       20.000000  9.818890e+05  2.518250e+04  8.210000e+02   2.478000e+03
75%       24.000000  3.683628e+06  1.140892e+05  3.357500e+03   9.241500e+03
max       43.000000  4.245389e+08  5.613827e+06  1.944971e+06   1.626501e+06

In [10]:
df_gb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 38916 entries, 0 to 38915
Data columns (total 16 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   video_id                38916 non-null  object
 1   trending_date           38916 non-null  object
 2   title                   38916 non-null  object
 3   channel_title           38916 non-null  object
 4   category_id             38916 non-null  int64 
 5   publish_time            38916 non-null  object
 6   tags                    38916 non-null  object
 7   views                   38916 non-null  int64 
 8   likes                   38916 non-null  int64 
 9   dislikes                38916 non-null  int64 
 10  comment_count           38916 non-null  int64 
 11  thumbnail_link          38916 non-null  object
 12  comments_disabled       38916 non-null  bool  
 13  ratings_disabled        38916 non-null  bool  
 14  video_error_or_removed  38916 non-null  bool  
 15  de

In [11]:
df_gb.head()

video_id trending_date  \
0  Jw1Y-zhQURU      17.14.11   
1  3s1rvMFUweQ      17.14.11   
2  n1WpP7iowLc      17.14.11   
3  PUTEiSjKwJU      17.14.11   
4  rHwDegptbI4      17.14.11   

                                               title  \
0      John Lewis Christmas Ad 2017 - #MozTheMonster   
1          Taylor Swift: …Ready for It? (Live) - SNL   
2         Eminem - Walk On Water (Audio) ft. Beyoncé   
3  Goals from Salford City vs Class of 92 and Fri...   
4  Dashcam captures truck's near miss with child ...   

                channel_title  category_id              publish_time  \
0                  John Lewis           26  2017-11-10T07:38:29.000Z   
1         Saturday Night Live           24  2017-11-12T06:24:44.000Z   
2                  EminemVEVO           10  2017-11-10T17:00:03.000Z   
3  Salford City Football Club           17  2017-11-13T02:30:38.000Z   
4            Cute Girl Videos           25  2017-11-13T01:45:13.000Z   

                                                tags     views   likes  \
0  christmas|"john lewis christmas"|"john lewis"|...   7224515   55681   
1  SNL|"Saturday Night Live"|"SNL Season 43"|"Epi...   1053632   25561   
2  Eminem|"Walk"|"On"|"Water"|"Aftermath/Shady/In...  17158579  787420   
3  Salford City FC|"Salford City"|"Salford"|"Clas...     27833     193   
4                                             [none]      9815      30   

   dislikes  comment_count                                  thumbnail_link  \
0     10247           9479  https://i.ytimg.com/vi/Jw1Y-zhQURU/default.jpg   
1      2294           2757  https://i.ytimg.com/vi/3s1rvMFUweQ/default.jpg   
2     43420         125882  https://i.ytimg.com/vi/n1WpP7iowLc/default.jpg   
3        12             37  https://i.ytimg.com/vi/PUTEiSjKwJU/default.jpg   
4         2             30  https://i.ytimg.com/vi/rHwDegptbI4/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   
3              False             False                   False   
4              False             False                   False   

                                         description  
0  Click here to continue the story and make your...  
1  Musical guest Taylor Swift performs …Ready for...  
2  Eminem's new track Walk on Water ft. Beyoncé i...  
3  Salford drew 4-4 against the Class of 92 and F...  
4  Dashcam captures truck's near miss with child ...

In [12]:
df_gb.nunique()

video_id                   3272
trending_date               205
title                      3369
channel_title              1625
category_id                  16
publish_time               3250
tags                       3122
views                     38393
likes                     30556
dislikes                  11092
comment_count             15778
thumbnail_link             3272
comments_disabled             2
ratings_disabled              2
video_error_or_removed        2
description                3704
dtype: int64

In [13]:
df_gb.duplicated().sum()

171

Here we see there are 171 duplicated rows in the Great Britain dataset

In [14]:
df_gb_duplicated = df_gb[df_gb.duplicated() == True]
df_gb_duplicated.head()
# df_gb[df_gb['ID'].duplicated() == True]

video_id trending_date  \
33743  RUCXD3_wW2w      18.14.05   
33744  p8npDG2ulKQ      18.14.05   
33745  aixso4N2vhI      18.14.05   
33746  bu0m_UdtoaU      18.14.05   
33747  i-G1hy73Mb8      18.14.05   

                                                   title  \
33743                                New Hulu Show - SNL   
33744  BTS (방탄소년단) LOVE YOURSELF 轉 Tear 'Singularity'...   
33745  President Trump Gives Remarks on the Joint Com...   
33746           Serious Questions: Avengers Infinity War   
33747  Eurovision Song Contest 2018 - Opening Ceremon...   

                 channel_title  category_id              publish_time  \
33743      Saturday Night Live           24  2018-05-13T04:59:28.000Z   
33744                  ibighit           10  2018-05-06T15:00:02.000Z   
33745          The White House           25  2018-05-08T18:45:37.000Z   
33746           Screen Junkies            1  2018-05-06T17:00:03.000Z   
33747  Eurovision Song Contest           24  2018-05-06T19:46:34.000Z   

                                                    tags     views    likes  \
33743  SNL|"Saturday Night Live"|"SNL Season 43"|"SNL...    723495     6362   
33744          BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"  26912663  2636004   
33745                                             [none]     89715     2806   
33746  screenjunkies|"screen junkies"|"serious questi...    443621    12892   
33747  Eurovision Song Contest|"2018"|"Lisbon"|"Openi...    698827    11721   

       dislikes  comment_count  \
33743      1589            420   
33744     27675         366899   
33745       488           1657   
33746      3829           2275   
33747       461           1175   

                                       thumbnail_link  comments_disabled  \
33743  https://i.ytimg.com/vi/RUCXD3_wW2w/default.jpg              False   
33744  https://i.ytimg.com/vi/p8npDG2ulKQ/default.jpg              False   
33745  https://i.ytimg.com/vi/aixso4N2vhI/default.jpg              False   
33746  https://i.ytimg.com/vi/bu0m_UdtoaU/default.jpg              False   
33747  https://i.ytimg.com/vi/i-G1hy73Mb8/default.jpg              False   

       ratings_disabled  video_error_or_removed  \
33743             False                   False   
33744             False                   False   
33745             False                   False   
33746             False                   False   
33747             False                   False   

                                             description  
33743  From the executive producer of Sex and the Cit...  
33744  BTS (방탄소년단) LOVE YOURSELF 轉 Tear 'Singularity'...  
33745                                    The White House  
33746  After watching Avengers: Infinity War we have ...  
33747  At 18:00 CEST today, the Blue Carpet ceremony ...

In [15]:
# confirming duplication using one row
df_gb.loc[df_gb['views'] == 723495]

video_id trending_date                title        channel_title  \
33629  RUCXD3_wW2w      18.14.05  New Hulu Show - SNL  Saturday Night Live   
33743  RUCXD3_wW2w      18.14.05  New Hulu Show - SNL  Saturday Night Live   

       category_id              publish_time  \
33629           24  2018-05-13T04:59:28.000Z   
33743           24  2018-05-13T04:59:28.000Z   

                                                    tags   views  likes  \
33629  SNL|"Saturday Night Live"|"SNL Season 43"|"SNL...  723495   6362   
33743  SNL|"Saturday Night Live"|"SNL Season 43"|"SNL...  723495   6362   

       dislikes  comment_count  \
33629      1589            420   
33743      1589            420   

                                       thumbnail_link  comments_disabled  \
33629  https://i.ytimg.com/vi/RUCXD3_wW2w/default.jpg              False   
33743  https://i.ytimg.com/vi/RUCXD3_wW2w/default.jpg              False   

       ratings_disabled  video_error_or_removed  \
33629             False                   False   
33743             False                   False   

                                             description  
33629  From the executive producer of Sex and the Cit...  
33743  From the executive producer of Sex and the Cit...

In [16]:
# removing the duplicated values and confirming
df_gb.drop_duplicates(inplace=True)
df_gb.duplicated().sum()

0

### United States

In [17]:
df_us = pd.read_csv("../input/youtube-new/USvideos.csv")
df_us.head()

video_id trending_date  \
0  2kyS6SvSYSE      17.14.11   
1  1ZAPwfrtAFY      17.14.11   
2  5qpjK5DgCt4      17.14.11   
3  puqaWrEC7tY      17.14.11   
4  d380meD0W0M      17.14.11   

                                               title          channel_title  \
0                 WE WANT TO TALK ABOUT OUR MARRIAGE           CaseyNeistat   
1  The Trump Presidency: Last Week Tonight with J...        LastWeekTonight   
2  Racist Superman | Rudy Mancuso, King Bach & Le...           Rudy Mancuso   
3                   Nickelback Lyrics: Real or Fake?  Good Mythical Morning   
4                           I Dare You: GOING BALD!?               nigahiga   

   category_id              publish_time  \
0           22  2017-11-13T17:13:01.000Z   
1           24  2017-11-13T07:30:00.000Z   
2           23  2017-11-12T19:05:24.000Z   
3           24  2017-11-13T11:00:04.000Z   
4           24  2017-11-12T18:01:41.000Z   

                                                tags    views   likes  \
0                                    SHANtell martin   748374   57527   
1  last week tonight trump presidency|"last week ...  2418783   97185   
2  racist superman|"rudy"|"mancuso"|"king"|"bach"...  3191434  146033   
3  rhett and link|"gmm"|"good mythical morning"|"...   343168   10172   
4  ryan|"higa"|"higatv"|"nigahiga"|"i dare you"|"...  2095731  132235   

   dislikes  comment_count                                  thumbnail_link  \
0      2966          15954  https://i.ytimg.com/vi/2kyS6SvSYSE/default.jpg   
1      6146          12703  https://i.ytimg.com/vi/1ZAPwfrtAFY/default.jpg   
2      5339           8181  https://i.ytimg.com/vi/5qpjK5DgCt4/default.jpg   
3       666           2146  https://i.ytimg.com/vi/puqaWrEC7tY/default.jpg   
4      1989          17518  https://i.ytimg.com/vi/d380meD0W0M/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   
3              False             False                   False   
4              False             False                   False   

                                         description  
0  SHANTELL'S CHANNEL - https://www.youtube.com/s...  
1  One year after the presidential election, John...  
2  WATCH MY PREVIOUS VIDEO ▶ \n\nSUBSCRIBE ► http...  
3  Today we find out if Link is a Nickelback amat...  
4  I know it's been a while since we did this sho...

In [18]:
# def compare_cols(df1, df2, df3):
#     if (df1.columns == df2.columns and df1.columns == df3.columns):
#         return True
#     else:
#         return False
# print(compare_cols(df_ca, df_gb, df_us))
df_ca.columns == df_us.columns

array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True])

Here we see that the columns in df_ca and df_us are the same

In [19]:
df_us.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40949 entries, 0 to 40948
Data columns (total 16 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   video_id                40949 non-null  object
 1   trending_date           40949 non-null  object
 2   title                   40949 non-null  object
 3   channel_title           40949 non-null  object
 4   category_id             40949 non-null  int64 
 5   publish_time            40949 non-null  object
 6   tags                    40949 non-null  object
 7   views                   40949 non-null  int64 
 8   likes                   40949 non-null  int64 
 9   dislikes                40949 non-null  int64 
 10  comment_count           40949 non-null  int64 
 11  thumbnail_link          40949 non-null  object
 12  comments_disabled       40949 non-null  bool  
 13  ratings_disabled        40949 non-null  bool  
 14  video_error_or_removed  40949 non-null  bool  
 15  de

In [20]:
df_us.describe()

category_id         views         likes      dislikes  comment_count
count  40949.000000  4.094900e+04  4.094900e+04  4.094900e+04   4.094900e+04
mean      19.972429  2.360785e+06  7.426670e+04  3.711401e+03   8.446804e+03
std        7.568327  7.394114e+06  2.288853e+05  2.902971e+04   3.743049e+04
min        1.000000  5.490000e+02  0.000000e+00  0.000000e+00   0.000000e+00
25%       17.000000  2.423290e+05  5.424000e+03  2.020000e+02   6.140000e+02
50%       24.000000  6.818610e+05  1.809100e+04  6.310000e+02   1.856000e+03
75%       25.000000  1.823157e+06  5.541700e+04  1.938000e+03   5.755000e+03
max       43.000000  2.252119e+08  5.613827e+06  1.674420e+06   1.361580e+06

In [21]:
df_us.nunique()

video_id                   6351
trending_date               205
title                      6455
channel_title              2207
category_id                  16
publish_time               6269
tags                       6055
views                     40478
likes                     29850
dislikes                   8516
comment_count             13773
thumbnail_link             6352
comments_disabled             2
ratings_disabled              2
video_error_or_removed        2
description                6901
dtype: int64

In [22]:
df_us.duplicated().sum()

48

We have 48 duplicated columns in the us dataset.

In [23]:
df_us[df_us.duplicated() == True].head()

video_id trending_date  \
34899  QBL8IRJ5yHU      18.15.05   
34900  t4pRQ0jn23Q      18.15.05   
34901  j4KvrAUjn6c      18.15.05   
34902  MAjY8mCTXWk      18.15.05   
34903  xhs8tf1v__w      18.15.05   

                                                   title     channel_title  \
34899  Why I'm So Scared (being myself and crying too...     grav3yardgirl   
34900  YoungBoy Never Broke Again Goes Sneaker Shoppi...           Complex   
34901         WE MADE OUR MOM CRY...HER DREAM CAME TRUE!  Lucas and Marcus   
34902  周杰倫 Jay Chou【不愛我就拉倒 If You Don't Love Me, It's...   杰威爾音樂 JVR Music   
34903  Terry Crews Answers the Web's Most Searched Qu...             WIRED   

       category_id              publish_time  \
34899           26  2018-05-14T19:00:01.000Z   
34900           24  2018-05-14T14:00:03.000Z   
34901           24  2018-05-13T18:03:56.000Z   
34902           10  2018-05-14T15:59:47.000Z   
34903           24  2018-05-14T16:00:29.000Z   

                                                    tags    views   likes  \
34899  beauty|"how to"|"makeup"|"howto"|"style"|"fash...  1469627  188652   
34900  sneakerhead|"complex"|"complex originals"|"sne...  1199587   49709   
34901  Lucas and Marcus|"Marcus and Lucas"|"Dobre"|"D...  3906727   77378   
34902  周杰倫|"Jay"|"Chou"|"周董"|"周杰伦"|"周傑倫"|"杰威尔"|"周周"|"...   916128   40485   
34903  autocomplete|"deadpool 2"|"google autocomplete...   343967   16988   

       dislikes  comment_count  \
34899      3124          33032   
34900      2380           7261   
34901     12160          15874   
34902      1042           4746   
34903       132           1308   

                                       thumbnail_link  comments_disabled  \
34899  https://i.ytimg.com/vi/QBL8IRJ5yHU/default.jpg              False   
34900  https://i.ytimg.com/vi/t4pRQ0jn23Q/default.jpg              False   
34901  https://i.ytimg.com/vi/j4KvrAUjn6c/default.jpg              False   
34902  https://i.ytimg.com/vi/MAjY8mCTXWk/default.jpg              False   
34903  https://i.ytimg.com/vi/xhs8tf1v__w/default.jpg              False   

       ratings_disabled  video_error_or_removed  \
34899             False                   False   
34900             False                   False   
34901             False                   False   
34902             False                   False   
34903             False                   False   

                                             description  
34899  I will never be able to say Thank You enough.....  
34900  YoungBoy Never Broke Again goes Sneaker Shoppi...  
34901  BEST MOM EVER! WANT TO SEE US IN NYC & NJ?!BUY...  
34902  詞：周杰倫、宋健彰（彈頭）  曲：周杰倫憂鬱型男的走心旋律  用英式搖滾宣洩情傷不愛我就拉倒...  
34903  Terry Crews takes the WIRED Autocomplete Inter...

In [24]:
# removing duplicates and confirming
df_us.drop_duplicates(inplace=True)
df_us.duplicated().sum()

0

### Merging the data

In [25]:
df = df_ca.append([df_gb, df_us])

In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 120527 entries, 0 to 40948
Data columns (total 16 columns):
 #   Column                  Non-Null Count   Dtype 
---  ------                  --------------   ----- 
 0   video_id                120527 non-null  object
 1   trending_date           120527 non-null  object
 2   title                   120527 non-null  object
 3   channel_title           120527 non-null  object
 4   category_id             120527 non-null  int64 
 5   publish_time            120527 non-null  object
 6   tags                    120527 non-null  object
 7   views                   120527 non-null  int64 
 8   likes                   120527 non-null  int64 
 9   dislikes                120527 non-null  int64 
 10  comment_count           120527 non-null  int64 
 11  thumbnail_link          120527 non-null  object
 12  comments_disabled       120527 non-null  bool  
 13  ratings_disabled        120527 non-null  bool  
 14  video_error_or_removed  120527 non-nu

In [27]:
df.duplicated().sum()

9133

In [28]:
# df[df.duplicated() == True].sort_values(by=df.video_id).head()